src/main/kotlin/no/skatteetaten/aurora/gobo/graphql/credentials/CredentialMutation.kt

```diff
@@ -40,6 +48,39 @@ class CredentialMutation(
                 resourceKind = ResourceKind.PostgresDatabaseInstance
             )
         ).toRegisterPostgresResult(input.host)
+            .sendNotification()
```

Comment:
Jeg må innrømme at jeg ikke er sånn suuuper fan av denne typen nestede metodekall i samarbeid med extensionmetoder - spesielt når de wrapper et .also-kall (eller lignende, run, f.eks).

Problemet er at det er ikke mulig å lese fra kallkjeden hvilke metode som returnerer hvilken type, og hvilke objekt som er mottaker av neste metodekall i kjeden. Dersom man flytter .also ut av extensionmetoden blir det litt tydligere;

```
        return herkimerService.registerResourceAndClaim(
            RegisterResourceAndClaimCommand(
                ownerId = dbhAdId,
                credentials = postgresInstance,
                resourceName = postgresInstance.instanceName,
                claimName = "ADMIN",
                resourceKind = ResourceKind.PostgresDatabaseInstance
            )
        ).toRegisterPostgresResult(input.host)
            .also { it.sendNotification() }
```

Da slipper man returnere noe fra sendNotification, samtidig som det er mulig å lese fra kallkjeden (via .also) hvilket objekt og type som blir returnert fra registerPostgresMotelServer uten å se på signaturen til registerPostgresMotelServer eller sendNotification. Personlig ville jeg ikke wrappet .also, siden den i praksis er laget for å slippe å eksplisitt lage chainable functions, samtidig som det blir lettere å lese koden uten å scrolle opp og ned eller bruke hjelpeverktøy.

I tillegg, igjen mine personlige preferanser, så hadde jeg utformet extensionmetodene og sendNotification-kallet slik

```
.also { nagHubService.sendNotification(it.toMessage()) }
```
siden RegisterPostgresResult.sendNotification() faktisk wrapper to contexter (RegisterPostgresResult og nagHubService). Det blir, etter min mening, litt tydeligere hva som skjer når man bare prøver lese registerPostgresMotelServer-metoden. Om du velger å beholde sendNotification på RegisterPostgresResult ville jeg uansett laget RegisterPostgresResult.toMessage()-extension.